Terminal.Gui/Windows/DefaultFileOperations.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Terminal.Gui.Resources;
+
+namespace Terminal.Gui {
+	/// <summary>
+	/// Default file operation handlers using modal dialogs.
+	/// </summary>
+	public class DefaultFileOperations : IFileOperations {
+		
+		/// <inheritdoc/>
+		public bool Delete (IEnumerable<FileSystemInfo> toDelete)
+		{
+			// Default implementation does not allow deleting multiple files
+			if (toDelete.Count () != 1) {
+				return false;
+			}
+			var d = toDelete.Single ();
+			var adjective =  d.Name;
+
+			int result = MessageBox.Query (
+				string.Format (Strings.fdDeleteTitle, adjective),
+				string.Format (Strings.fdDeleteBody, adjective),
+				Strings.fdYes, Strings.fdNo);
+
+			try {
+				if (result == 0) {
+					if (d is FileInfo) {
+						d.Delete ();
+					} else {
+						((DirectoryInfo)d).Delete (true);
+					}
+
+                    return true;
+				}
+			} catch (Exception ex) {
+				MessageBox.ErrorQuery (Strings.fdDeleteFailedTitle, ex.Message, "Ok");
+			}
+
+            return false;
+		}
+
+        private bool Prompt(string title,string defaultText, out string result)
```

Comment:
Why not use MessageBox here?

Reply:
MessageBox doesn't support text input does it?

![image](https://user-images.githubusercontent.com/31306100/227381309-f903c654-8142-4cbb-af83-be2ec7fbafe5.png)